opm/autodiff/AquiferCarterTracy.hpp

```diff
@@ -0,0 +1,505 @@
+                   c2_ ; //6.283 (METRIC, PVT-M); 1.1191 (FIELD); 6.283 (LAB).
+
+            // Variables for influence table
+            std::vector<Scalar> aqutab_td_, aqutab_pi_;
+
+            // Cumulative flux
+            Scalar W_flux_, dt_, pa0_, gravity_;
+
+            // Also return the polynomial fit
+            std::vector<Scalar> coeff_;
+            
+
+            // We fit the tabular data using a polynomial fit
+            // Modified from Copyright (C) 2014  Clifford Wolf <clifford@clifford.at> 
+            // http://svn.clifford.at/handicraft/2014/polyfit/polyfit.cc
+            inline void polynomial_fit( const std::vector<Scalar> &X, const std::vector<Scalar> &y, 
+                                        std::vector<Scalar> &coeff, int order, bool bias) const
+            {
+                int colNum = (bias)? order + 1 : order;
+                Eigen::MatrixXd A(X.size(), colNum);
+                Eigen::VectorXd y_mapped = Eigen::VectorXd::Map(&y.front(), y.size());
+                Eigen::VectorXd result;
+
+                assert(X.size() == y.size());
+                assert(X.size() >= colNum);
+
+                // create matrix
+                for (size_t i = 0; i < X.size(); i++)
+                for (size_t j = 0; j < colNum; j++)
+                    A(i, j) = (bias)? pow(X.at(i), j) : pow(X.at(i), j+1);
+
+                // solve for linear least squares fit
+                result = A.householderQr().solve(y_mapped);
+
+                coeff.resize(colNum);
+                for (size_t i = 0; i < colNum; i++)
+                    coeff[i] = result[i];
+            }
+
+            inline void init_quantities(const Aquancon::AquanconOutput& connection)
+            {
+                // We reset the cumulative flux at the start of any simulation, so, W_flux = 0
+                W_flux_ = 0.;
+
+                // We next get our connections to the aquifer and initialize these quantities using the initialize_connections function
+                initialize_connections(connection);
+
+                // pa0_ is the initial aquifer water pressure. Must be calculated from equilibrium if left default,
+                // or we get the information from the deck Hacked to make it at 45e6 Pa
+                calculate_reservoir_equilibrium();
+                
+                pressure_previous_.resize(cell_idx_.size(), 0.);
+                pressure_current_.resize(cell_idx_.size(), 0.);
+
+                rhow_.resize(cell_idx_.size(), 998.0); 
+                Qai_.resize(cell_idx_.size(), 0.);
+
+                polynomial_fit(aqutab_td_, aqutab_pi_, coeff_, 2, true);
+            }
+
+            inline void get_current_Pressure_cell(std::vector<Scalar>& pressure_water, const int idx, const IntensiveQuantities& intQuants)
+            {
+                const auto& fs = intQuants.fluidState();
+                pressure_water[idx] = fs.pressure(FluidSystem::waterPhaseIdx).value();
```

Comment:
If you think you are solving the Aquifer related in a implicit way, you should probably keep the derivatives here, which makes the pressure_water be the variable of type of `Eval`.  Please keep in mind, with the current local AD, the derivatives are against the primary variables related to the this specific cell only, they should not interact with derivatives from other cells.                                                                     

Solving with explicit way might work as long as it gets converged.